<a href="https://colab.research.google.com/github/opoluliakh/data-analytics-learning/blob/main/Poluliakh_HW_%D0%9F%D0%B5%D1%80%D0%B5%D0%B2%D1%96%D1%80%D0%BA%D0%B0_%D1%81%D1%82%D0%B0%D1%82%D0%B8c%D1%82%D0%B8%D1%87%D0%BD%D0%B8%D1%85_%D0%B3%D1%96%D0%BF%D0%BE%D1%82%D0%B5%D0%B7_Z_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Перевірка статиcтичних гіпотез. Z-test.



0. Зчитайте дані з `data.csv` у змінну data, яка має тип pandas.DataFrame. Ми будемо далі працювати з цією змінною.

In [2]:
import pandas as pd

data = pd.read_csv("/data.csv")

data.head()


Unnamed: 0       data
0           0  17.499453
1           1  19.662399
2           2   7.182823
3           3  29.841625
4           4   9.239386

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного `numpy.ndarray` вектора та мати розмірність (100,). Така розмірність означає, що дані - одновимірні (колонка), якщо б розмірність була (100, 1), то дані сприймаються методами як двовимірні (таблиця), хоча для нас виглядати можуть так само.

In [3]:
data = data.data.values
data.shape

(100,)

**Завдання 1**. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2025 року.

До цього місяця, ми мали продажі в середньому на чек $20.

Необхідно зʼясувати, чи вийшло у нас статистично значущо **підвищити** середній чек за липень 2025?

Аби дати відповідь - ми проведемо z-test на рівні значущості $\alpha=0.05$ двома способами. В цьому завданні ми будемо виконувати обчислення "вручну" з використанням бібліотек numpy та scipy.stats подібно до прикладу в лекції. Для цього виконайте наступні 7 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть параметри задачі у змінні Python:
    - `sample_mean` - $\bar{x}$, середнє значення в вибірці
    - `population_mean` - $\mu_0$, середнє значення в популяції (тобто те, з яким ми порівнюємо середнє вибіркове значення)
    - `population_std` - $\sigma$, вибіркове стандартне відхилення, яке ми вважаємо, що є рівним ст. відх. популяції, адже маємо достатньо велику вибірку.
    - `sample_size` - $n$, розмір вибірки
    - `alpha` - $\alpha$ рівень значущості
    
    **Увага!** Для обчислення стандартного відхилення маємо скористатись функцією `np.std(your_dataframe, ddof=1)`. Чому так - розбираємо в лекції про t-test.


2. Визначте для цієї задачі:
    - якою є гіпотеза $H_0$
    - якою є альтернативна гіпотеза $H_a$
    - з яким типом тесту ми маємо справу - лівосторонній, правосторонній чи двосторонній.

3. Обчисліть стандартну помилку SE.
4. Розрахуйте z-статистику.
5. Знайдіть критичне z-значення з допомогою бібліотеки stats.
6. Обчисліть p-value з допомогою бібліотеки stats.
7. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$. Для прийняття рішення зробіть і порівняння z-статистики з критичним значенням, і проаналізуйте p-value.

In [4]:
## Завдання 1. Перевірка, чи середній чек > 20 доларів у липні 2025

### 1. Параметри задачі

import numpy as np
from scipy import stats

sample_mean = np.mean(data)
population_mean = 20
population_std = np.std(data, ddof=1)
sample_size = len(data)
alpha = 0.05

sample_mean, population_std, sample_size


(np.float64(19.37755319670441), np.float64(6.25356123240434), 100)

In [5]:
## 2


SE = population_std / np.sqrt(sample_size)


z_stat = (sample_mean - population_mean) / SE

SE, z_stat


(np.float64(0.625356123240434), np.float64(-0.995347738933505))

In [7]:
## 3

z_crit = stats.norm.ppf(1 - alpha)

p_value = 1 - stats.norm.cdf(z_stat)

z_crit, p_value


(np.float64(1.6448536269514722), np.float64(0.8402164165464012))

In [8]:
## 4

if (z_stat >= z_crit) and (p_value < alpha):
    print("Відхиляємо H0: середній чек статистично значуще зріс.")
else:
    print("Не відхиляємо H0: доказів зростання середнього чеку недостатньо.")


Не відхиляємо H0: доказів зростання середнього чеку недостатньо.


**Завдання 2.** Виконайте обчислення z-test з використанням бібліотеки statsmodels.

Отримайте z-статистику та р-значення.

Виведіть p-значення та зробіть висновок, чи ми маємо достатньо доказів, аби стверджувати, що середній чек зріс.

Чи зійшлись значення z-статистику та р-значення в цьому завданні з попередніми обчисленнями?

Завдання оцінюється в 3 бали.

In [10]:
## Завдання 2

from statsmodels.stats.weightstats import ztest
import numpy as np

mu0 = 20
alpha = 0.05

z_stat_sm, p_value_sm = ztest(data, value=mu0, alternative='larger')

print("Z-статистика (statsmodels):", z_stat_sm)
print("p-value (statsmodels):", p_value_sm)

if p_value_sm < alpha:
    print("Відхиляємо H0: середній чек зріс.")
else:
    print("Не відхиляємо H0: доказів зростання недостатньо.")



Z-статистика (statsmodels): -0.9953477389335053
p-value (statsmodels): 0.8402164165464012
Не відхиляємо H0: доказів зростання недостатньо.


### Висновок і порівняння з попереднім завданням
Очікуємо **Z ≈ -0.995** і **p-value ≈ 0.84** (бо середнє < 20, а тест правосторонній).
Рішення однакове: **не відхиляємо $H_0$** на рівні $\alpha=0.05$ — статистичних доказів зростання середнього чеку немає.
